hack/grab-profiles.sh

```diff
@@ -0,0 +1,149 @@
+#!/bin/bash
+
+# Copyright 2015 The Kubernetes Authors All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+SERVER_ADDR=""
+REQUESTED_PROFILES=""
+MEM_PPROF_FLAGS=""
+PROFILE_PATHS=""
+PROFILE_COMPONENTS=""
+OUTPUT_DIR="."
+HEAPSTER_VERSION="v0.18.2"
+TUNNEL_PORT="${TUNNEL_PORT:-1234}"
+
+ARGS=$(getopt -o s:mho: -l server:,output:,inuse-space,inuse-objects,alloc-space,alloc-objects,cpu,master,heapster -- "$@")
+MASTER_PPROF_PATH="debug/pprof"
+HEAPSTER_PPROF_PATH="api/v1/proxy/namespaces/kube-system/services/monitoring-heapster/debug/pprof"
+
+if [ $? -ne 0 ]; then
+  echo "Wrong argument"
+  exit 1
+fi
+
+eval set -- "$ARGS"
+
+while true; do
+  case $1 in
+    -s|--server)
+      shift
+      if [ -z "$1" ]; then
+        echo "empty argument to --server flag"
+        exit 1
+      fi
+      SERVER_ADDR=$1
+      shift
+      ;;
+    -m|--master)
+      shift
+      PROFILE_COMPONENTS="master $PROFILE_COMPONENTS"
+      ;;
+    -h|--heapster)
+      shift
+      PROFILE_COMPONENTS="heapster $PROFILE_COMPONENTS"
+      ;;
+    -o|--output)
+      shift
+      if [ -z "$1" ]; then
+        echo "empty argument to --output flag"
+        exit 1
+      fi
+      OUTPUT_DIR=$1
+      shift
+      ;;
+    --inuse-space)
+      shift
+      REQUESTED_PROFILES="mem $REQUESTED_PROFILES"
+      MEM_PPROF_FLAGS="inuse_space $MEM_PPROF_FLAGS"
+      ;;
+    --inuse-objects)
+      shift
+      REQUESTED_PROFILES="mem $REQUESTED_PROFILES"
+      MEM_PPROF_FLAGS="inuse_objects $MEM_PPROF_FLAGS"
+      ;;
+    --alloc-space)
+      shift
+      REQUESTED_PROFILES="mem $REQUESTED_PROFILES"
+      MEM_PPROF_FLAGS="alloc_space $MEM_PPROF_FLAGS"
+      ;;
+    --alloc-objects)
+      shift
+      REQUESTED_PROFILES="mem $REQUESTED_PROFILES"
+      MEM_PPROF_FLAGS="alloc_objects $MEM_PPROF_FLAGS"
+      ;;
+    --cpu)
+      shift
+      REQUESTED_PROFILES="cpu"
+      ;;
+    --)
+      shift
+      break;
+      ;;
+  esac
+done
+
+if [ -z "$SERVER_ADDR" ]; then
+  echo "Server flag is required"
+  exit 1
+fi
+
+if [ -z "$PROFILE_COMPONENTS" ]; then
+  echo "Choose at least one component to profile"
+  exit 1
+fi
+
+gcloud compute ssh $SERVER_ADDR --ssh-flag=-nN --ssh-flag=-L${TUNNEL_PORT}:localhost:8080 &
+
+# Watt for ssh tunnel to be created
+sleep 10
+
+SSH_PID=$(ps aux | grep "1234:localhost:8080" | grep ssh | tr -s " " | cut -f2 -d" ")
+trap 'kill $SSH_PID' EXIT
```

Comment:
We can, but in the end it's the same. Plus killing in this way will work on non-GCE providers, so people adapting this script to their needs will have a slightly less things to worry about.